opencog/nlp/relex2logic/rule-helpers.scm

```diff
@@ -1056,6 +1110,74 @@
 		)
 	))])
 )
+
+;-----------------------------------------------------------------------
+; complement clauses
+;-----------------------------------------------------------------------
+(define (complement-rule comp_concept comp_instance pred_concept pred_instance)
+	(list 
+		(ImplicationLink (PredicateNode comp_instance df-node-stv) (PredicateNode comp_concept df-node-stv) df-link-stv)
+		(ImplicationLink (PredicateNode pred_instance df-node-stv) (PredicateNode pred_concept df-node-stv) df-link-stv)
+		(EvaluationLink df-link-stv
+			(PredicateNode comp_instance df-node-stv)
+			(ListLink df-link-stv
+				(PredicateNode pred_instance df-node-stv)
+			)
+		)
+	)
+)
+
+(define (compmod-rule comp_concept comp_instance pred_concept pred_instance)
+	(list 
+		(ImplicationLink (PredicateNode comp_instance df-node-stv) (PredicateNode comp_concept df-node-stv) df-link-stv)
+		(ImplicationLink (PredicateNode pred_instance df-node-stv) (PredicateNode pred_concept df-node-stv) df-link-stv)
+		(EvaluationLink df-link-stv
+			(PredicateNode "InManner" df-node-stv)
+			(ListLink df-link-stv
+				(PredicateNode pred_instance df-node-stv)
+				(ConceptNode comp_instance df-node-stv)
+			)
+		)
+	)
+)
+
+(define (because-rule comp_concept comp_instance pred_concept pred_instance)
+	(list 
+		(ImplicationLink (PredicateNode comp_instance df-node-stv) (PredicateNode comp_concept df-node-stv) df-link-stv)
+		(ImplicationLink (PredicateNode pred_instance df-node-stv) (PredicateNode pred_concept df-node-stv) df-link-stv)
+		(EvaluationLink df-link-stv
+			(PredicateNode "Because" df-node-stv)
+			(ListLink df-link-stv
+				(PredicateNode pred_instance df-node-stv)
+				(ConceptNode comp_instance df-node-stv)
+			)
+		)
+	)
+)
+
+(define (attime-rule comp_concept comp_instance pred_concept pred_instance)
+	(list 
+		(ImplicationLink (PredicateNode comp_instance df-node-stv) (PredicateNode comp_concept df-node-stv) df-link-stv)
+		(ImplicationLink (PredicateNode pred_instance df-node-stv) (PredicateNode pred_concept df-node-stv) df-link-stv)
+		(AtTimeLink df-link-stv)
+			(PredicateNode pred_instance df-node-stv)
```

Comment:
shouldn't there be a TimeNode? http://wiki.opencog.org/w/TimeServer#Temporal_association_in_AtomSpace_hypergraph_and_TimeServer